src/anvato/AnvatoDrmFairplayContentProtectionIntegration.web.ts

```diff
@@ -0,0 +1,97 @@
+import type { CertificateRequest, ContentProtectionIntegration, LicenseRequest, MaybeAsync } from 'react-native-theoplayer';
+import {
+  fromBase64StringToUint8Array,
+  fromObjectToUint8Array,
+  fromUint8ArrayToBase64String,
+  fromUint8ArrayToObject,
+  LicenseResponse,
+} from 'react-native-theoplayer';
+import type { AnvatoDrmConfiguration } from './AnvatoDrmConfiguration';
+import type { AnvatoLicenseRequestError } from './AnvatoLicenseRequestError';
+
+export class AnvatoDrmFairplayContentProtectionIntegration implements ContentProtectionIntegration {
+  static readonly DEFAULT_CERTIFICATE_URL = 'insert default certificate url here';
+  static readonly DEFAULT_LICENSE_URL = 'insert default license url here';
+
+  private readonly contentProtectionConfiguration: AnvatoDrmConfiguration;
+  private contentId: string | undefined = undefined;
+
+  constructor(configuration: AnvatoDrmConfiguration) {
+    this.contentProtectionConfiguration = configuration;
+  }
+
+  onCertificateRequest(request: CertificateRequest): MaybeAsync<Partial<CertificateRequest> | BufferSource> {
+    request.url =
+      this.contentProtectionConfiguration.fairplay?.certificateURL ?? AnvatoDrmFairplayContentProtectionIntegration.DEFAULT_CERTIFICATE_URL;
+    request.headers = {
+      ...request.headers,
+      'content-type': 'application/octet-stream',
+    };
+    return request;
+  }
+
+  async onLicenseRequest(request: LicenseRequest): Promise<BufferSource> {
+    const url = (request.url =
+      this.contentProtectionConfiguration.fairplay?.licenseAcquisitionURL ?? AnvatoDrmFairplayContentProtectionIntegration.DEFAULT_LICENSE_URL);
+    const spcMessage = fromUint8ArrayToBase64String(request.body!);
+    const bodyObject = {
+      spc: spcMessage,
+      assetid: this.contentId,
+    };
+
+    // Do license request
+    const response = await fetch(url, {
+      method: request.method,
+      headers: {
+        ...request.headers,
+        'Content-Type': 'application/octet-stream',
+      },
+      body: fromObjectToUint8Array(bodyObject),
+    });
+
+    // Something went wrong, try to get detailed info by parsing the response body.
+    if (response.status >= 400) {
+      try {
+        const jsonResult: AnvatoLicenseRequestError = await response.json();
+        const error = jsonResult.error;
+
+        // Extract drmErrorCode from error message
+        const drmErrorCode = extractDrmErrorCode(error.message);
+        return Promise.reject({
```

Comment:
Nit pick: prefer `throw` inside `async` functions.

Also, consider rejecting with an actual `Error` instead of a plain object. That way, you'll get a proper stack trace. 🙂